### DISCUSSION TITLE
How to create nested pxe/grub menu

### DISCUSSION THREAD

**Discussion by chris1984** [cobbler/cobbler#3246] [Q&A]
Hi,

I installed Cobbler 3.2.2 and I imported a few distros, how do I add a folder for those to keep them organized.

**Comment by SchoolGuy** [comment]
So when you have distros you need to add profiles to actually have something you could use. After that you can optionally (but ideally) add systems and then Cobbler should automatically organize that in folders for you.

**Reply by chris1984** [reply]
@SchoolGuy We don't really use systems, on our 2.x setup we just had distros and that created profiles, when the user pxe booted on that it would run but we didn't have any systems setup. 

Can we still use nested menus without systems or does it have to use systems? We just want distros/profiles. We wanted to Organize all the different RHEL 7.x 8.x releases into a folder called RHEL7 RHEL8

**Reply by chris1984** [reply]
![screenshot-vcenter toledo satellite lab eng rdu2 redhat com-2022 08 29-15_35_15](https://user-images.githubusercontent.com/1518655/187696935-b363d48e-f0fb-4383-a879-073bbb53957b.png)

![nested2](https://user-images.githubusercontent.com/1518655/187697279-84c47717-57fe-436d-8ac1-c6e385947c5b.png)

Here is how our eng group has it laid out, inside of that they just have profiles for server/desktop/workstation etc

**Reply by SchoolGuy** [reply]
You can use profiles for sure. Just try it out in a dev environment. But you need profiles, without them it will not work.

**Comment by marbaa** [comment]
Regarding menu. 
1. Could someone please explain to me in short difference in PXE menu between 2.x and 3.x? In Cobbler 2.x there is this nice blue menu. In Cobbler 3.x there is simple text on black background only. Why?
2. I have one profile, and multiple systems created.

![image](https://user-images.githubusercontent.com/40859847/198889817-c8ec99d7-a058-4881-a14e-1b47bfa6054b.png)

But when I PXE boot, I see only one entry there:
- systems have configured IP, MAC and hostname
![image](https://user-images.githubusercontent.com/40859847/198889840-954ec9ab-3102-41ea-a127-ad34e6fa8f8a.png)

3. Why I see also Profilename in boot menu?
4. If I select Profilename I get into sub menu where is again Profilename.

![image](https://user-images.githubusercontent.com/40859847/198889872-d46a510c-c601-4281-b14c-d1b0c3bf238c.png)


```
[root@home-cobbler ~]# cat /etc/almalinux-release
AlmaLinux release 8.6 (Sky Tiger)
[root@home-cobbler ~]# cobbler --version
Cobbler 3.2.2
  source: ?, ?
  build time: Fri Aug 26 01:42:52 2022
```

**Reply by SchoolGuy** [reply]
So what you see is the GRUB menu. I am too young (literally) to have seen the blue menu but I believe that was pxelinux back in the day.

In regard to the structure:

1. You see first the local boot & the distros
2. In the submenu of the distro you see the profiles
3. The reason why they have the same name is because you probably have used `cobbler import` and then the profiles have the same name as the distros.

Other open questions?
